go-runtime/compile/schema.go

```diff
@@ -106,44 +107,34 @@ type ParseResult struct {
 	Errors []*schema.Error
 }
 
-// ExtractModuleSchema statically parses Go FTL module source into a schema.Module.
-func ExtractModuleSchema(dir string, sch *schema.Schema) (optional.Option[ParseResult], error) {
+func legacyExtractModuleSchema(dir string, sch *schema.Schema, out *analyzers.ExtractResult) error {
```

Comment:
got a bit hairy attempting to run the legacy extractor as a completely isolated process since things are so entangled

instead modified the legacy code to accept an output pointer as param (`out *analyzers.ExtractResult`). the initial caller provides the partial result from the prior extraction step, which gives the legacy extractor access to details (native names and decls) that it needs. the output/result is updated in-place during the legacy extraction